src/v/storage/tests/utils/random_batch.cc

```diff
@@ -81,18 +87,24 @@ model::record_batch make_random_batch(
   model::offset o,
   int num_records,
   bool allow_compression,
-  model::record_batch_type bt) {
+  model::record_batch_type bt,
+  std::optional<std::vector<size_t>> record_sizes) {
```

Comment:
nit: it seems like `std::optional<std::vector<size_t>>` could also be `std::vector<size_t>` where an empty vector is treated the same as `std::nullopt`?

Reply:
I just wanted to make this explicit in the code. If the behaviour is triggered by the empty vector it's not that clear.